api/tests/src/task/notifications/test_opportunity_notification.py

```diff
@@ -4,178 +4,279 @@
-    opportunity = factories.OpportunityFactory.create(no_current_summary=True)
-    summary = factories.OpportunitySummaryFactory.create(
-        opportunity=opportunity,
-        close_date=datetime_util.utcnow() + timedelta(days=21),
-    )
-    factories.CurrentOpportunitySummaryFactory.create(
-        opportunity=opportunity, opportunity_summary=summary
-    )
-    saved_opportunity = factories.UserSavedOpportunityFactory.create(
-        user=user,
-        opportunity=opportunity,
-        last_notified_at=opportunity.updated_at - timedelta(days=1),
-    )
-
-    factories.OpportunityChangeAuditFactory.create(
-        opportunity=opportunity,
-        updated_at=saved_opportunity.last_notified_at + timedelta(minutes=1),
-    )
-
-    # Run the notification task
-    task = EmailNotificationTask(db_session, search_client)
-    task.run()
-
-    # Verify notification log was created
-    notification_logs = db_session.query(UserNotificationLog).all()
-    assert len(notification_logs) == 1
-
-    log = notification_logs[0]
-    assert log.user_id == user.user_id
-    assert log.notification_reason == NotificationReason.OPPORTUNITY_UPDATES
-
-    assert log.notification_sent is True
-
-
-def test_no_notification_log_when_no_updates(
-    cli_runner,
-    db_session,
-    search_client,
-    enable_factory_create,
-    user,
-    user_with_email,
-):
-    """Test that no notification log is created when there are no updates"""
-    # Create a saved opportunity that doesn't need notification
-    opportunity = factories.OpportunityFactory.create(no_current_summary=True)
-    summary = factories.OpportunitySummaryFactory.create(
-        opportunity=opportunity,
-        close_date=datetime_util.utcnow() + timedelta(days=21),
-    )
-    factories.CurrentOpportunitySummaryFactory.create(
-        opportunity=opportunity, opportunity_summary=summary
-    )
-    factories.UserSavedOpportunityFactory.create(
-        user=user,
-        opportunity=opportunity,
-        last_notified_at=opportunity.updated_at + timedelta(minutes=1),  # After the update
-    )
-
-    # Run the notification task
-    task = EmailNotificationTask(db_session, search_client)
-    task.run()
-
-    # Verify no notification log was created
-    notification_logs = db_session.query(UserNotificationLog).all()
-    assert len(notification_logs) == 0
+class TestOpportunityNotification:
+
+    @pytest.fixture
+    def email_notification_task(self, db_session, search_client, monkeypatch):
+        monkeypatch.setenv("AWS_PINPOINT_APP_ID", "test-app-id")
+        monkeypatch.setenv("FRONTEND_BASE_URL", "http://testhost:3000")
+        monkeypatch.setenv("ENABLE_OPPORTUNITY_NOTIFICATIONS", "true")
+        monkeypatch.setenv("ENABLE_SEARCH_NOTIFICATIONS", "false")
+        monkeypatch.setenv("ENABLE_CLOSING_DATE_NOTIFICATIONS", "false")
+
+        config = EmailNotificationConfig()
+        return EmailNotificationTask(db_session, search_client, config)
```

Comment:
For these tests, why not run the OpportunityNotificationTask directly rather than the general email task? Would be less configuration necessary.

Also note that you can just pass the values into the config instead of setting env vars first, you just have to use the alias names like:

```py
config = EmailNotificationConfig(AWS_PINPOINT_APP_ID="whatever")
```

Reply:
Checking on this again